src/NexusMods.Collections/Verbs.cs

```diff
@@ -47,8 +47,18 @@ private static async Task<int> InstallCollection([Injected] IRenderer renderer,
 
         var revisionMetadata = await nexusModsLibrary.GetOrAddCollectionRevision(collectionFile, CollectionSlug.From(slug), RevisionNumber.From((ulong)revision), token);
 
-        await collectionDownloader.DownloadItems(revisionMetadata, itemType: CollectionDownloader.ItemType.Required, db: connection.Db, cancellationToken: token);
-
+        try
+        {
+            await collectionDownloader.DownloadItems(revisionMetadata, itemType: CollectionDownloader.ItemType.Required, db: connection.Db,
+                cancellationToken: token
+            );
+        }
+        catch (OperationCanceledException e)
+        {
+            await renderer.Error(e.Message);
+            return 0;
+        }
```

Comment:
The code that you added to the `CollectionDownloader` regarding getting the links should be added here.

You'd wanna check with the `ILoginManager` whether the user is premium or not, if they are you can use `collectionDownloader.DownloadItems`, if they are a free user then you should print all download links.